src/pages/Asset.vue

```diff
@@ -481,10 +484,51 @@ export default {
   },
   methods: {
     async afterAuth () {
+      let afterAuthFirstTime
+      if (this.afterAuthFirstTime) {
+        afterAuthFirstTime = this.afterAuthFirstTime
+        this.afterAuthFirstTime = false
+      }
+
       // Not blocking. Move to (blocking) preFetch and remove $route watcher if optimizing for SEO
       // with server-side rendering (SSR)
       this.fetchRelatedAssets()
       this.fetchAssetRatingsByTransaction()
+
+      const routeQuery = this.$route.query
+
+      if (afterAuthFirstTime && this.stripeActive) {
```

Comment:
The role of `afterAuthFirstTime` is not clearly revealed by its name.

Plus I wonder whether it’s a good idea to include so much payment code in `Asset.vue` file while some forks may want to ignore payments.

Reply:
I moved the code into a `payment` mixin to not pollute `Asset.vue` logic.

`afterAuthFirstTime` is ill-named. It was there to trigger the payment logic once because `afterAuth()` is triggered twice at app load.

Reply:
It is triggered twice due to in mounted hook in some mixin and to currentUser watch right (probably after the user is set)?

In this case you could just check the current user to prevent redundant calls.

Reply:
Yes, I'll trigger the payment logic only if the current user is authenticated.